clients/cpp/CPPSketch/json_hub_protocol.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+#ifndef _JSONHUBPROTOCOL_H
+#define _JSONHUBPROTOCOL_H
+
+#include <tuple>
+#include <string>
+#include <sstream>
+
+#include "rapidjson\document.h"
+#include "rapidjson\writer.h"
+#include "rapidjson\stringbuffer.h"
+
+#include "transfer_format.h"
+
+namespace signalR
+{
+    template <typename T>
+    rapidjson::Value to_json(T item, rapidjson::MemoryPoolAllocator<>& alloc);
+
+    template <typename T>
+    T from_json(const rapidjson::Value& item);
+
+    class json_hub_protocol
+    {
+    public:
+        template <typename ...T>
+        std::tuple<T...> parse_message(const std::string& data) const
```

Comment:
This is obviously going to need some work still. We need to return parsed versions of all the messages, support non-text protocols via some kind of byte buffer input, etc.